### ISSUE TITLE
Vectors use different mask type on scalar platforms

### ISSUE THREAD

**Issue by makspll** [bitshifter/glam-rs#436]
Hi, I've been facing some issues in my crate which performs some codegen assuming the API of any sub-crates doesn't change depending on the platform: https://github.com/makspll/bevy_mod_scripting/issues/70

However it appears that in this library, `Vec4` returns either a `BVec4A` or `BVec4` for many methods depending on which platform you compile, on x86 this is `BVec4A` whereas on the Mac m2 Arm 64 it seems to be `BVec4A` 

Ultimately this logic is captured in the codegen:

```jinja
{% if is_scalar %}
    {% set mask_t = "BVec" ~ dim %}
{% else %}
    {% set is_simd = true %}
    {% if is_sse2 %}
        {% set simd_t = "__m128" %}
    {% elif is_wasm32 %}
        {% set simd_t = "v128" %}
    {% elif is_coresimd %}
        {% set simd_t = "f32x4" %}
    {% endif %}
    {% set mask_t = "BVec" ~ dim ~ "A" %}
{% endif %}
```

This means my generated code fails to compile on Mac. Is this intentional? Why does the `BVec4A` type not simply reduce to a `BVec4` type functionally (not alias just different implementation depending on if simd is supported) ?

**Comment by makspll** [comment]
Linux arm64 builds face a similar issue

**Comment by bitshifter** [comment]
It is intentional, but I recognize that it causes problems. Primarily it does this because if you don't have SIMD then Vec4A will perform worse than Vec4.

Were you able to work around it? Possibly the inconsistency is worse than the performance loss, I could look at changing it, it doesn't come up a whole lot though (that comes to my attention anyway).

Longer term I am considering some larger scale changes to glam that will mean this goes away but I want to collect my thoughts on it and consult with the community before doing anything.

**Comment by makspll** [comment]
Currently I have a PR which generates a second file for scalar platforms, essentially causing scripts to be platform dependent. A proper fix will require me to essentially reduce the API to the lowest common denominator or write custom types which hide these differences. Ultimately it's a massive pain to fix but it *is* possible.

**Comment by bitshifter** [comment]
Fixed in #442.
